ledger/src/blockstore_meta.rs

```diff
@@ -406,6 +406,29 @@ impl ErasureMeta {
     }
 }
 
+#[allow(dead_code)]
+impl MerkleRootMeta {
+    pub(crate) fn from_shred(shred: &Shred) -> Self {
+        Self {
+            merkle_root: shred.merkle_root().unwrap_or_default(),
```

Comment:
Now I am thinking that `MerkleRootMeta.merkle_root` should have been `Option<Hash>` to explicitly distinguish legacy shreds.
This `unwrap_or_default` here is a bit hacky, and it seems like we will have to special case `merkle_root_meta.merkle_root == Hash::default()` later.

Reply:
definitely possible, however I was under the impression that legacy shreds are no longer being broadcasted.
I assume at some point you will delete `shred/legacy.rs` at which point I can remove `unwrap_or_default()` and any `Hash::default()` checks.

Reply:
They are no longer broadcasted but still totally valid to send legacy shreds.
We might also need to keep `shred/legacy.rs` for backward compatibility.
I am not sure for how long we will need to preserve legacy shreds, so I am leaning to use `Option<Hash>` to be consistent with current state of the code.

Reply:
Got it. Is it supported to send a mixture of legacy and merkle shreds in a single FEC set?
If not I think we can just avoid storing the merkle root meta for legacy FEC sets. There's no benefit to inserting the meta into blockstore if the FEC set only has legacy shreds. For the next pr that writes this column, I can filter out FEC sets that are legacy.

Reply:
no, but if you receive two shreds with the same `fec_set_index` and one is merkle and the other legacy, then that is a proof of duplicate block.
So I think we need to record that we have received a legacy shred for this `fec_set_index`, and I guess the easiest way to do that is to set `merkle_root` to `None`

Reply:
makes sense i'll do that first.